src/plots/cartesian/category_mode_defaults.js

```diff
@@ -0,0 +1,39 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+module.exports = function handleCategoryModeDefaults(containerIn, containerOut, coerce) {
+
+    if(containerIn.type === 'category') {
+
+        var validCategories = ['trace', 'category ascending', 'category descending', 'array'];
```

Comment:
we should `require('./layout_attrbitues')` into this files and have:

```js
var validCategories = layoutAttributes.categorymode.values;
```

to keep thing DRY :palm_tree: